assets/tests/components/detours/diversionPage.test.tsx

```diff
@@ -90,7 +90,82 @@ describe("DiversionPage", () => {
     expect(screen.queryByTitle("Detour End")).not.toBeInTheDocument()
   })
 
-  test("directions start with origin intersection when second waypoint is added", async () => {
+  test("has a normal cursor before placing the start point", async () => {
+    const { container } = render(<DiversionPage />)
+
+    await waitFor(() => {
+      expect(
+        container.querySelector(".c-detour_map--map__clickable")
+      ).not.toBeInTheDocument()
+    })
+  })
+
+  test("has a waypoint cursor, given by the __clickable class, once the start point is placed", async () => {
+    const { container } = render(<DiversionPage />)
+
+    act(() => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+
+    await waitFor(() => {
+      expect(
+        container.querySelector(".c-detour_map--map__clickable")
+      ).toBeInTheDocument()
+    })
+  })
+
+  test("has no waypoints at the beginning or when placing the start point", async () => {
+    const { container } = render(<DiversionPage />)
+
+    expect(
+      container.querySelectorAll(".c-detour_map-circle-marker--detour-point")
+    ).toHaveLength(0)
+
+    act(() => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+
+    await waitFor(() => {
+      expect(
+        container.querySelectorAll(".c-detour_map-circle-marker--detour-point")
+      ).toHaveLength(0)
+    })
+  })
+
+  test("can click on map to add a waypoint", async () => {
+    const { container } = render(<DiversionPage />)
+
+    act(() => {
+      fireEvent.click(originalRouteShape.get(container))
+    })
+    act(() => {
+      fireEvent.click(container.querySelector(".c-vehicle-map")!)
+    })
+
+    await waitFor(() => {
+      expect(
+        container.querySelectorAll(".c-detour_map-circle-marker--detour-point")
```

Comment:
I think a lot of these tests could be cleaned up quite a lot with more judicious use of [test selectors](https://github.com/mbta/skate/blob/eb1aa60ba2f73eb5d227eee61c07823635491f59/assets/tests/testHelpers/selectors/components/detours/diversionPage.tsx)

Then this, for instance, could read

```tsx
espect(waypointMarker.getAll(container)).toHaveLength(1)
```

The more boilerplate, like these hard-coded selectors, we get out of the meat of the tests, the easier they'll be to work with, and the more reliable their coverage will be.

---

I'd like to do that, but I'd like to do it as a follow-up task.

Reply:
https://app.asana.com/0/1152340551558956/1207337422166095/f